src/async.rs

```diff
@@ -447,3 +452,23 @@ impl AsyncClient {
 fn is_status_retryable(status: reqwest::StatusCode) -> bool {
     RETRYABLE_ERROR_CODES.contains(&status.as_u16())
 }
+
+/// Trait that defines the ability to sleep within an async runtime
+#[async_trait]
+pub trait Sleeper {
+    /// Wait until the specified `duration` has elapsed
+    async fn sleep(duration: Duration);
+}
```

Comment:
I think the way to go is not rely on `async_trait`, and use the associated type as I mentioned in the comments, [here's a reference](https://github.com/bitcoindevkit/rust-esplora-client/pull/107/files#diff-db4e074f116ae66822e8ace789be86c3c01ffc1659520910bd702b22687d0ff6R454-R470).